pkgs/top-level/all-packages.nix

```diff
@@ -419,16 +419,15 @@ with pkgs;
 
   deviceTree = callPackage ../os-specific/linux/device-tree { };
 
-  octodns = python3Packages.callPackage ../tools/networking/octodns { };
+  octodns = callPackage ../tools/networking/octodns { };
 
   octodns-providers = recurseIntoAttrs {
-    bind = python3Packages.callPackage ../tools/networking/octodns/providers/bind { };
-    gandi = python3Packages.callPackage ../tools/networking/octodns/providers/gandi { };
-    hetzner = python3Packages.callPackage ../tools/networking/octodns/providers/hetzner { };
-    powerdns = python3Packages.callPackage ../tools/networking/octodns/providers/powerdns { };
-    cloudflare = python3Packages.callPackage ../tools/networking/octodns/providers/cloudflare { };
-    ddns = python3Packages.callPackage ../tools/networking/octodns/providers/ddns { };
-    transip = python3Packages.callPackage ../tools/networking/octodns/providers/transip { };
+    bind = octodns.python.pkgs.callPackage ../tools/networking/octodns/providers/bind { };
+    gandi = octodns.python.pkgs.callPackage ../tools/networking/octodns/providers/gandi { };
+    hetzner = octodns.python.pkgs.callPackage ../tools/networking/octodns/providers/hetzner { };
+    powerdns = octodns.python.pkgs.callPackage ../tools/networking/octodns/providers/powerdns { };
+    cloudflare = octodns.python.pkgs.callPackage ../tools/networking/octodns/providers/cloudflare { };
+    ddns = octodns.python.pkgs.callPackage ../tools/networking/octodns/providers/ddns { };
   };
```

Comment:
You could put this attribute set into the package passthru and only reference it here so you are closer to the required `callPackage` scope.

Additionally, you could use simply implement autocalling instead.

```
    plugins = lib.recurseIntoAttrs (
      lib.packagesFromDirectoryRecursive {
        inherit (python.pkgs) callPackage;
        directory = ./plugins;
      }
    );